features/publish.feature

```diff
@@ -101,6 +101,20 @@ Feature: Publish reports
       │ module.exports = { default: '--publish-quiet' }                          │
       └──────────────────────────────────────────────────────────────────────────┘
       """
+
+  @spawn
+  Scenario: when results are not published due to an error raised by the server, the banner is displayed
+    When I run cucumber-js with env `CUCUMBER_PUBLISH_TOKEN=keyboardcat`
+    Then it fails
```

Comment:
As a general point, not sure how users would feel about their CI pipeline failing because e.g. cucumber reports is down or they got their token wrong (but all tests passed). We could wait for feedback and look at making it configurable via formatter options if needed.

Reply:
Well, that's already the case if you publish with a wrong token, `cucumber-js` exits with code 1. What this PR does is just providing extra information.
And I think it should fail if the publication did not succeed, as something went wrong. It a bit like generating a JUnit report but the output directory can't be written, you would expect the run to exit with an error no ?

Reply:
I agree with @vincent-psarga 
People who expect reports to be published from a CI for example should be noticed that something went wrong.

I run a few tests with cucumber-ruby. I was expecting a non-zero exit code, it was not the case. I think I'll open an issue in cucumber-ruby about that.

I will try the same with cucumber-jvm

Reply:
Makes sense to me. I think it’s fine to kick the option down the road and see if there’s demand.